numba/np/npyfuncs.py

```diff
@@ -648,11 +648,13 @@ def np_complex_exp_impl(context, builder, sig, args):
 def np_real_exp2_impl(context, builder, sig, args):
     _check_arity_and_homogeneity(sig, args, 1)
 
-    ll_ty = args[0].type
-    fnty = llvmlite.ir.FunctionType(ll_ty, [ll_ty,])
-    fn = cgutils.insert_pure_function(builder.module, fnty,
-                                      name='llvm.exp2')
-    return builder.call(fn, [args[0]])
+    dispatch_table = {
+        types.float32: 'numba_exp2f',
+        types.float64: 'numba_exp2',
+    }
+
+    return _dispatch_func_by_name_type(context, builder, sig, args,
+                                       dispatch_table, 'exp2')
```

Comment:
Should we split the impl depending on the platform? Are we disabling potential optimization (vectorization) on non-windows?

Reply:
I think this will potentially limit optimisation as it generates 'unknown' library calls. However, with a "minimising risk for 0.57.0 release" view, I think perhaps it should be left as written as it's essentially doing the same as what was shipped in 0.56.4 (deferred symbol calls). Am happy to consider views from other folks that are interested, the patch is small and the changes to accommodate a split implementation are also small.

Reply:
Ok, minimal risk is a good reason